Updated readme.md file (#37)
dotnet/corefx-testdata: README.md

# CoreFx-TestData

This repository contains test assets that are binary files or too large to be checked into corefx directly. Packages are produced and uploaded to the dotnet blob feed: https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json.

Uploading the packages is currently a manual step. Community members can submit PRs but the deployment needs to be done by a .NET team member as described below.

## Workflow

1. Modify the \*.csproj project file(s) and increment the version number.
2. Generate the nuget package file for your new test data using `dotnet pack`.

> 3. Test that corefx can import the new nuget package file.

Reviewer: I would add a `[Optional but encouraged]` on step 3 and 4, as they are not really necessary parts of the workflow. Technically speaking, the necessary parts are steps 1, 2, 5, 6, and 7.

Author: Done

> 3. Test that corefx can import the new nuget package file.

Author: ```suggestion
3. (Optional) Test that corefx can import the new nuget package file.
```

> 4. Verify that you can consume the new test assets in your corefx unit tests.

Author: ```suggestion
4. (Optional) Verify that you can consume the new test assets in your corefx unit tests.
```

5. Submit a corefx-testdata PR with your new test assets and bumped version project file.

> 6. Upload the nuget package file to the dotnet blob feed.

Reviewer: I know I told you not to do this, but perhaps it is better to have the link https://github.com/dotnet/core-eng/tree/master/Documentation/Tools/dotnet-core-push-oneoff-package in here for now. If the core-eng team has a problem with this we can find a better way to redirecting to that.

Author: Done

> 6. Upload the nuget package file to the dotnet blob feed.

Author: ```suggestion
6. Upload the nuget package file to the dotnet blob feed (only .NET team members can do this). See: https://github.com/dotnet/core-eng/tree/master/Documentation/Tools/dotnet-core-push-oneoff-package
```

7. Submit a corefx PR with your unit test changes that consume the new nuget package.

## Example

We are going to make the following assumptions:

> - We cloned the `dotnet/corefx-testdata` GitHub project in `D:\corefx-testdata`.

Reviewer: NIT: `GitHub project` -> `GitHub repo`

> - We cloned the `dotnet/corefx-testdata` GitHub project in `D:\corefx-testdata`.

Author: ```suggestion
- We cloned the `dotnet/corefx-testdata` GitHub repo in `D:\corefx-testdata`.
```

- We cloned the `dotnet/corefx` GitHub project in `D:\corefx`.

> - We are working on adding a new unit test for the GZip feature from the `System.IO.Compression` namespace, and the test depends on a file called `example.gz`.

Reviewer: In order for your example to work, there is one more assumption that you must make, which is that people have run build.cmd in the corefx repo. If you don't do that, then you won't be able to run step 3.

Author: Right, I forgot to mention that. Thanks.
```suggestion
- We are working on adding a new unit test for the GZip feature from the `System.IO.Compression` namespace, and the test depends on a file called `example.gz`.
- We already compiled corefx with `build.cmd`.
```

Open the `corefx\external\test-runtime\XUnit.Runtime.depproj` file, find the `<PackageReference>` element for your TestData project and change the version to the same number you used in step 1.

```xml
<!-- Test Data -->
<PackageReference Include="System.IO.Compression.TestData" Version="1.0.10" />
```

Now from the root of the `corefx` project, run the commands that will import the package file:

```
PS D:\corefx> .\.dotnet\dotnet.exe restore .\external\test-runtime\XUnit.Runtime.depproj -s D:\corefx-testdata\System.IO.Compression.TestData\bin\Debug\
    Restore completed in 1.08 sec for D:\corefx\external\test-runtime\XUnit.Runtime.depproj.

PS D:\corefx> .\.dotnet\dotnet.exe msbuild .\external\test-runtime\XUnit.Runtime.depproj /t:rebuild

    Microsoft (R) Build Engine version 16.1.67-preview+g13843078ee for .NET Core
    Copyright (C) Microsoft Corporation. All rights reserved.

    Restore completed in 292.24 ms for D:\corefx\external\test-runtime\XUnit.Runtime.depproj.

     C:\Users\username\.nuget\packages\...
     C:\Users\username\.nuget\packages\...
     ...
```

>  Now check if the nuget package was exported correctly. Go to your current user's nuget packages folder, usually located in `C:\Users\yourusername\.nuget\packages`. Find the folder for your TestData and inside you will find the original dependencies plus the new ones you added:

Reviewer: NIT: `was exported` -> `was restored`

Reviewer: NIT: `C:\Users\yourusername\` -> `%UserProfile%`

Reviewer: NIT: `you will find the original dependencies plus the new ones you added:` -> `you should see a new folder with the new version of the package that you produced.`

Author: Fixed

>  Now check if the nuget package was exported correctly. Go to your current user's nuget packages folder, usually located in `C:\Users\yourusername\.nuget\packages`. Find the folder for your TestData and inside you will find the original dependencies plus the new ones you added:

Author: ```suggestion
 Now check if the nuget package was restored correctly. Go to your current user's nuget packages folder, usually located in `%UserProfile%\.nuget\packages`. Find the folder for your TestData and inside you should see a new folder with the new version of the package you produced:
```

```

> PS D:\corefx> dir C:\Users\yourusername\.nuget\packages\System.IO.Compression.TestData\1.0.10\

Author: ```suggestion
PS D:\corefx> dir %UserProfile%\.nuget\packages\System.IO.Compression.TestData\1.0.10\
```

    Directory: C:\Users\yourusername\.nuget\packages\System.IO.Compression.TestData\1.0.10

Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----        3/25/2019   3:55 PM                content
-a----        3/25/2019   3:55 PM            130 .nupkg.metadata
-a----        3/25/2019   3:55 PM       54372842 system.io.compression.testdata.1.0.10.nupkg
-a----        3/25/2019   3:55 PM             88 system.io.compression.testdata.1.0.10.nupkg.sha512
-a----        3/25/2019   3:55 PM           1034 system.io.compression.testdata.nuspec

> PS D:\corefx> dir C:\Users\yourusername\.nuget\packages\System.IO.Compression.TestData\1.0.10\content\GZipTestData\

Author: ```suggestion
PS D:\corefx> dir $Env:UserProfile\.nuget\packages\System.IO.Compression.TestData\1.0.10\content\GZipTestData\
```

    Directory: C:\users\yourusername\.nuget\packages\System.IO.Compression.TestData\1.0.10\content\GZipTestData

Mode                LastWriteTime         Length Name
----                -------------         ------ ----

> -a----        3/25/2019  3:55 PM           1748 example.gz

Author: ```suggestion
-a----        3/25/2019  3:55 PM           1748 example.gz
\`\`\`
```

> 

Reviewer: Looks like you missed a ``` here before the ####4

Author: I'll add it separately, I don't know how to comment those characters as code suggestions.

> 

Author: ```

# CoreFx-TestData

> This repository contains test assets that are binary file or too large to be checked into corefx directly. Packages are produced and uploaded to the [dotnet blob feed](https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json).

Reviewer: NIT: `binary file` -> `binary files`

Reviewer: Also, keeping the feed in parenthesis rather than linking to it might be better. The reason is that the link will take you to the index.json which doesn't really tell you much info, and usually, when following these steps you would want to see the feed you are publishing to in the instructions instead.

> This repository contains test assets that are binary file or too large to be checked into corefx directly. Packages are produced and uploaded to the [dotnet blob feed](https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json).

Author: ```suggestion
This repository contains test assets that are binary files or too large to be checked into corefx directly. Packages are produced and uploaded to the [dotnet blob feed](https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json).
```

Uploading the packages is currently a manual step. Community members can submit PRs but the deployment needs to be done by a .NET team member as described below.

## Workflow

For the examples provided in the steps below, we are going to make the following assumptions:

- We cloned the `dotnet/corefx-testdata` GitHub project in `D:\corefx-testdata`.
- We cloned the `dotnet/corefx` GitHub project in `D:\corefx`.
- We are working on adding a new unit test for the GZip feature from the `System.IO.Compression` namespace, and the test depends on a file called `example.gz`.

#### 1. Modify the project files and increment the version number.

We decided to save our `example.gz` file in `corefx-testdata\System.IO.Compression.TestData\GZipTestData\example.gz`.

Now we edit the `corefx-testdata\System.IO.Compression.TestData\System.IO.Compression.TestData.csproj` file and bump the patch fragment of the version number.

```xml
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <PackageVersion>1.0.10</PackageVersion>
    </PropertyGroup>
</Project>
```

> *Note: if the version was 1.0.9, bump it to 1.0.10, not 1.1.0.*

Reviewer: I would remove this. Above you already mentioned that the part of the version to bump is the patch version, which is the number after the second dot. If anything, I would just add a parenthesis note after mentioning the word patch above specifying this is the third part of the version.

Author: Do you mean remove the example?

Reviewer: nope, just remove the *Note
